media-cdn/snippets.py

```diff
@@ -0,0 +1,217 @@
+    policy = policy_pattern.format(
+            encoded_url_prefix=encoded_url_prefix,
+            expires=expiration_timestamp,
+            key_name=key_name)
+
+    digest = ed25519.Ed25519PrivateKey.from_private_bytes(
+        decoded_key).sign(policy.encode('utf-8'))
+    signature = base64.urlsafe_b64encode(digest).decode('utf-8')
+    signature = signature[:-2]  # to remove trailing '=' symbols
+    signed_url = u'{url}{separator}{policy}&Signature={signature}'.format(
+            url=stripped_url,
+            separator='&' if query_params else '?',
+            policy=policy,
+            signature=signature)
+    return signed_url
+# [END mediacdn_sign_url]
+
+
+# [START mediacdn_sign_cookie]
+def sign_cookie(url_prefix, key_name, base64_key, expiration_time):
+    """Gets the Signed cookie value for the specified URL prefix and configuration.
+
+    Args:
+        url_prefix: URL prefix to sign as a string.
+        key_name: name of the signing key as a string.
+        base64_key: signing key as a base64 encoded string.
+        expiration_time: expiration time as a UTC datetime object.
+
+    Returns:
+        Returns the Cloud-CDN-Cookie value based on the specified configuration.
+    """
+    encoded_url_prefix = base64.urlsafe_b64encode(
+            url_prefix.strip().encode('utf-8')).decode('utf-8')
+    epoch = datetime.datetime.utcfromtimestamp(0)
+    expiration_timestamp = int((expiration_time - epoch).total_seconds())
+    decoded_key = base64.urlsafe_b64decode(base64_key)
+
+    policy_pattern = u'URLPrefix={encoded_url_prefix}:Expires={expires}:KeyName={key_name}'
+    policy = policy_pattern.format(
+            encoded_url_prefix=encoded_url_prefix,
+            expires=expiration_timestamp,
+            key_name=key_name)
+
+    digest = ed25519.Ed25519PrivateKey.from_private_bytes(
+        decoded_key).sign(policy.encode('utf-8'))
+    signature = base64.urlsafe_b64encode(digest).decode('utf-8')
+    signature = signature[:-2]  # remove trailing '='
+
+    signed_policy = u'Cloud-CDN-Cookie={policy}:Signature={signature}'.format(
+            policy=policy, signature=signature)
+    return signed_policy
+# [END mediacdn_sign_cookie]
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+            description=__doc__,
+            formatter_class=argparse.RawDescriptionHelpFormatter)
+
+    subparsers = parser.add_subparsers(dest='command')
+
+    sign_url_parser = subparsers.add_parser(
+            'sign-url',
+            help="Sign a URL to grant temporary authorized access.")
+    sign_url_parser.add_argument(
+            'url', help='The URL to sign.')
+    sign_url_parser.add_argument(
+            'key_name',
+            help='Key name for the signing key.')
+    sign_url_parser.add_argument(
+            'base64_key',
+            help='The base64 encoded signing key.')
+    sign_url_parser.add_argument(
+            'expiration_time',
```

Comment:
1. I think simple bash command like this will do
```bash
python snippets.py sign-url http://35.186.234.33/index.html my-key BxwXXNjeGaoWqjr7GHEymRJkP4SaOC12dTGixk7Yr8I= `expr $(date +%s) + 1200`
```
2. TTL does not accurately tell, till when a resource is valid. I see epoch start and end as much accurate way.

Reply:
As suggested by other-reviewers cleared CLI steps & arg-parse code